File: WoofAndroid/presentation/grid/src/main/kotlin/com/perrystreet/woof/presentation/grid/ui/preview/GridPreviewData.kt
package com.perrystreet.woof.presentation.grid.ui.preview

import com.perrystreet.woof.models.dog.Dog
import com.perrystreet.woof.presentation.grid.uimodel.DogCellUIModel
import com.perrystreet.woof.presentation.grid.viewmodel.GridViewModel

internal object GridPreviewData {
    fun loaded(): GridViewModel.State.Loaded = GridViewModel.State.Loaded(
        cells = List(12) { index ->
            val dog = Dog(id = index.toLong(), name = "Dog $index", photoUrl = "")
            DogCellUIModel(index = index, id = dog.id, name = dog.name, photoUrl = dog.photoUrl, domain = dog)
        },
        isLoadingMore = true,
        hasMore = true,
    )
}

File: WoofIOS/SwiftPackages/Presentation/Grid/Sources/PresentationGrid/UI/Preview/GridPreviewData.swift
import Models

enum GridPreviewData {
    static func loaded() -> GridViewModel.State {
        .loaded(
            GridViewModel.State.Loaded(
                cells: (0..<12).map { index in
                    let dog = Dog(id: index, name: "Dog \(index)", photoUrl: "")
                    return DogCellUIModel(index: index, id: dog.id, name: dog.name, photoUrl: dog.photoUrl, domain: dog)
                },
                isLoadingMore: true,
                hasMore: true
            )
        )
    }
}
